### PR TITLE
Fetch CLI verisons from Homebrew where applicable.

### LINKED ISSUE #3230: Latest version number should come from brew
Anytime I run a Pulumi command, I get this warning right now:

```
$ pulumi ...
warning: A new version of Pulumi is available. To upgrade from version '1.0.0' to '1.1.0', run
   $ brew upgrade pulumi
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

Yet running the command as suggested just issues an error:

```
Error: pulumi 64 already installed
```

I assume this is because the latest version hasn't been merged into Brew yet: https://github.com/pulumi/pulumi/issues/3221.

This is quite annoying as an end user. Is there any reason we can't take the latest version from Brew itself for purposes of checking if you're on the latest?

### PR DESCRIPTION
**Pull request by pgavlin**
If the CLI seems to have been installed using Homebrew, do not consult
the service for the latest version. Instead, consult the Homebrew JSON
API.

Fixes #3230.

### CODE REVIEW

**Comment by pgavlin** [comment]
The JSON API docs are here: https://formulae.brew.sh/docs/api/#get-formula-metadata-for-a-homebrew-core-formula

File: cmd/pulumi.go [hunk 1]
```diff
@@ -226,6 +229,14 @@ func getCLIVersionInfo() (semver.Version, semver.Version, error) {
 		return latest, oldest, err
 	}
 
+	brewLatest, isBrew, err := getLatestBrewFormulaVersion()
+	if err != nil {
+		return semver.Version{}, semver.Version{}, err
+	}
+	if isBrew {
+		return brewLatest, brewLatest, nil
```
Comment (justinvp) [inline cmd/pulumi.go]: Do we want to cache this information like we do when requesting it from our service?

Comment (pgavlin) [reply cmd/pulumi.go]: Yep--just fixed that :)

File: cmd/pulumi.go [hunk 2]
```diff
@@ -232,6 +235,15 @@ func getCLIVersionInfo() (semver.Version, semver.Version, error) {
 		return semver.Version{}, semver.Version{}, err
 	}
 
+	brewLatest, isBrew, err := getLatestBrewFormulaVersion()
```
Comment (justinvp) [inline cmd/pulumi.go]: Do we want to avoid calling our service if `isBrew`? Or is there a reason to always do it (e.g. for version usage telemetry)?

Comment (pgavlin) [reply cmd/pulumi.go]: The latter.

**Pull request merged**